services/hh/processor.go

```diff
@@ -144,8 +157,25 @@ func (p *Processor) Process() error {
 	p.mu.RLock()
 	defer p.mu.RUnlock()
 
-	res := make(chan error, len(p.queues))
-	for nodeID, q := range p.queues {
+	// Only process queues for active nodes.
+	var activeQueues map[uint64]*queue
```

Comment:
I think pulling this `activeQueues` logic into a `func activeQueues()` help keep the `Process` function simpler.